tests/test_types.py

```diff
@@ -473,7 +473,12 @@ def test_enum_fails():
     with pytest.raises(ValueError) as exc_info:
         CookingModel(tool=3)
     assert exc_info.value.errors() == [
-        {'loc': ('tool',), 'msg': 'value is not a valid enumeration member', 'type': 'type_error.enum'}
+        {
+            'loc': ('tool',),
+            'msg': 'value is not a valid enumeration member; permitted: 1, 2',
+            'type': 'type_error.enum',
+            'ctx': {'enum_type': ToolEnum},
```

Comment:
putting the type in the context (rather than the permitted values, as in WrongConstantError), means that the type is visible in the traceback for debugging purposes, while only the permitted values are visible in the error message (which may be served externally).

Reply:
makes sense, the idea of `ctx` was something like "think you might need when rendering your own error message".